### ISSUE TITLE
Fix S4056 FP: Pointless warnings on method invocations that are known to ignore the passed culture

### ISSUE THREAD

**Issue by bart-vmware** [SonarSource/sonar-dotnet#8233]
See https://github.com/SonarSource/sonar-dotnet/issues/6752#issuecomment-1768257155.

**Comment by sebastien-marichal** [comment]
Hello @bart-vmware,

Thank you for reporting this FP.

I agree with your comment, this gets quite annoying when coming from .NET6 to .NET 7 or .NET 8.
I valid this as an FP.

However, after a quick look, I was not able to see that [`System.Convert.ToInt32`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Convert.cs#L1091-L1096) is not using the culture, it seems to rely on it somehow. But once again, I only had a quick look.
If you can elaborate on this example it might help us.

Thank you,

----

### For implementation

A large amount of work will be required to identify all the types implementing the [`IParseable`](https://learn.microsoft.com/en-us/dotnet/api/system.iparsable-1) that ignore the culture.

Here are the few examples identified as such:
- [Guid.Parse](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Guid.cs#L1744)
- [Boolean.Parse](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Boolean.cs#L410)

This might apply to `TryParse` as well.

**Comment by martin-strecker-sonarsource** [comment]
Integer parsing is culture-dependent for e.g. the negative sign:
```cs
int.Parse("-1"); // OK
int.Parse("-1", CultureInfo.CreateSpecificCulture("fa-AF")); // Fails
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1659](https://sonarsource.atlassian.net/browse/NET-1659)

[NET-1659]: https://sonarsource.atlassian.net/browse/NET-1659?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by bart-vmware** [comment]
It's _impossible_ for me to view your new ticking system. Because I _can't_ create an account there. Whether I try to sign up using Google, Microsoft, or my company's email address, my company's SSO always kicks in, then informs me that I need to request access to _our internal_ Jira system to proceed.

What's the new way of creating and tracking issues? GitHub worked very well, sad to see it's being replaced.

**Comment by pavel-mikula-sonarsource** [comment]
Hi @bart-vmware,

We're not happy with this Jira experience either, but we had to move unfortunately a while ago.

The official way to report issue is to use the community forum, as guided by the "New issue" page. We're keeping this ticket open and will update it once a fix gets released (you know, unless we forget 😅). We're also working on re-enabling GH Issue creation in the near future for FPs/FNs.

**Comment by bart-vmware** [comment]
Why did you have to move?

**Comment by pavel-mikula-sonarsource** [comment]
Most of the company is Java-based and on Jira. So we had to start using that too. I'll delete this after sending as the message is not relevant for the issue

**Comment by pavel-mikula-sonarsource** [comment]
@bart-vmware GH issues are now reopened and it is possible to submit new ones again.

**Comment by bart-vmware** [comment]
That's nice, thanks!
